core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -0,0 +1,278 @@
+      case _ => Set.empty
+    }
+
+    shouldAllowEveryoneIfNoAclIsFound = if (config.contains(SimpleAclAuthorizer.AllowEveryoneIfNoAclIsFoundProp))
+      javaConfigs.get(SimpleAclAuthorizer.AllowEveryoneIfNoAclIsFoundProp).toString.toBoolean
+    else false
+
+    val zkUrl = config.getOrElse(SimpleAclAuthorizer.ZkUrlProp, kafkaConfig.zkConnect).toString
+    val zkConnectionTimeoutMs = config.getOrElse(SimpleAclAuthorizer.ZkConnectionTimeOutProp, kafkaConfig.zkConnectionTimeoutMs).toString.toInt
+    val zkSessionTimeOutMs = config.getOrElse(SimpleAclAuthorizer.ZkSessionTimeOutProp, kafkaConfig.zkSessionTimeoutMs).toString.toInt
+
+    zkClient = ZkUtils.createZkClient(zkUrl, zkConnectionTimeoutMs, zkSessionTimeOutMs)
+    ZkUtils.makeSurePersistentPathExists(zkClient, SimpleAclAuthorizer.AclZkPath)
+
+    aclChangeListener = new ZkNodeChangeNotificationListener(zkClient, SimpleAclAuthorizer.AclChangedZkPath, SimpleAclAuthorizer.AclChangedPrefix, AclChangedNotificaitonHandler)
+    aclChangeListener.startup()
+
+    zkClient.subscribeStateChanges(ZkStateChangeListener)
+
+    loadCache
+  }
+
+  override def authorize(session: Session, operation: Operation, resource: Resource): Boolean = {
+    val principal: KafkaPrincipal = new KafkaPrincipal(KafkaPrincipal.USER_TYPE, session.principal.getName)
+    val host = session.host
+
+    trace(s"Authorizer invoked to authorize principal = $principal from host = $host on resource = $resource for operation = $resource")
+
+    if (superUsers.contains(principal)) {
+      debug(s"principal = $principal is a super user, allowing operation without checking acls.")
+      return true
+    }
+
+    val acls: Set[Acl] = getAcls(resource)
+    if (acls.isEmpty) {
+      debug(s"No acl found for resource $resource , authorized = $shouldAllowEveryoneIfNoAclIsFound")
+      return shouldAllowEveryoneIfNoAclIsFound
+    }
+
+    //first check if there is any Deny acl match that would disallow this operation.
+    if (aclMatch(session, operation, resource, principal, host, Deny, acls))
+      return false
+
+    /**
+     * if principal is allowed to read or write we allow describe by default, the reverse does not apply to Deny.
+     */
+    val ops: Set[Operation] = if (Describe.equals(operation))
+      Set[Operation](operation, Read, Write)
+    else
+      Set[Operation](operation)
+
+    //now check if there is any allow acl that will allow this operation.
+    if (ops.exists(operation => aclMatch(session, operation, resource, principal, host, Allow, acls)))
+      return true
+
+    //We have some acls defined and they do not specify any allow ACL for the current session, reject request.
+    debug(s"principal = $principal from host = $host is not allowed to perform operation = $operation  on resource = $resource as no explicit acls were defined to allow the operation")
```

Comment:
I rechecked to assure it is logged in all cases (superuser, no acls, deny acl match(as part of matching function), allow acl match(as part of matching function),  no acl found).

If we want it to be logged at the end, I can change the logic so it does not return in middle. May be its just me, I think the current way looks cleaner.